# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mc-mp/umode_t.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

On some architectures it just build accidently, however on e.g. powerpc64
defining umode_t results in un-parse-able constructs in <asm/types.h>.
Just include it to get the define.

  - Rene Rebe <rene@exactcode.de>

--- mc-4.1.40-pre9/config.h.in.vanilla	2008-08-23 12:13:57.000000000 +0200
+++ mc-4.1.40-pre9/config.h.in	2008-08-23 12:14:43.000000000 +0200
@@ -141,7 +141,7 @@
 #undef STAT_STATVFS
 
 /* Define umode_t if your system does not provide it */
-#undef umode_t
+#include <asm/types.h>
 
 /* Define nlink_t if your system does not provide it */
 #undef nlink_t
